packages/menu/src/utils.ts

```diff
@@ -98,12 +98,6 @@ type ReducerAction = {
 export const stateReducer: Reducer<ReducerState, ReducerAction> = (state, action) => {
   let changes: ReducerState | null = null;
 
-  // Reset `focusOnOpen` if it was previously set to true; this prevents
-  // forced focus on the initial item in future state updates.
-  if (state.focusOnOpen) {
-    changes = { ...state, focusOnOpen: false };
-  }
-
```

Comment:
The `useEffect` hook resets `focusOnOpen` on update, eliminating the need for an explicit dispatch call. This method has proven to be more reliable, especially with nested submenus.